test/get_previous_releases.py

```diff
@@ -148,6 +148,24 @@ def download_binary(tag, args) -> int:
     ret = subprocess.run(['tar', '-zxf', tarball, '-C', tag,
                           '--strip-components=1',
                           'bitcoin-{tag}'.format(tag=tag[1:])]).returncode
+
+    if tag >= "v23" and platform == "arm64-apple-darwin":
+        # Starting with v23.0 there are arm64 binaries for ARM (M1/M2) macs, but they have to be signed to run
+        binary_path = f'{os.getcwd()}/{tag}/bin/bitcoind'
+
+        # Is it already signed?
+        result = subprocess.run(['codesign', '-v', '-d', binary_path], capture_output=True)
+        if "code object is not signed at all" in result.stderr.decode("utf-8"):
+            # Have to self-sign the binary
+            subprocess.run(['codesign', '-s', '-', binary_path], capture_output=True)
```

Comment:
This nicely returns 0 for me on Intel, but I haven't checked the failure mode.

Reply:
You're right. It returns 1 if the `-v` check doesn't find a signature, but the subsequent calls do return 0 (on successful self-sign and on verifying that new signature).

This will allow me to simplify the checks and not rely on the stderr messages. Update coming.